lib/Doctrine/DBAL/Platforms/SqlitePlatform.php

```diff
@@ -375,6 +373,31 @@ protected function _getCreateTableSQL($name, array $columns, array $options = []
         return $query;
     }
 
+    /**
+     * Generate a PRIMARY KEY definition if no autoincrement value is used
+     *
+     * @param array $columns
+     * @param array $options
+     *
+     * @return string
+     */
+    private function getNonAutoincrementPrimaryKeyDefinition(array $columns, array $options)
+    {
+        if (! isset($options['primary']) || empty($options['primary'])) {
+            return '';
+        }
+
+        $keyColumns = array_unique(array_values($options['primary']));
+
+        foreach ($keyColumns as $keyColumn) {
+            if (isset($columns[$keyColumn]['autoincrement']) && ! empty($columns[$keyColumn]['autoincrement'])) {
+                return '';
```

Comment:
Could you check if it's technically possible to have a PK which consists of multiple columns one or more of which has `AUTOINCREMENT`? If yes, then it would be a breaking change since in this case, the PK will consist only declared for the autoincremented column.

Reply:
That's currently not possible. It will result in a `MappingException` while generating the scheme:
"Single id is not allowed on composite primary key in entity...."

Reply:
Then we're not breaking anything. Sounds good.